azurerm/resource_arm_private_link_service_test.go

```diff
@@ -0,0 +1,676 @@
+					resource.TestCheckResourceAttr(resourceName, "visibility_subscription_ids.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "visibility_subscription_ids.0"),
+					resource.TestCheckResourceAttr(resourceName, "nat_ip_configuration.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "nat_ip_configuration.0.private_ip_address", "10.5.1.17"),
+					resource.TestCheckResourceAttr(resourceName, "nat_ip_configuration.0.private_ip_address_version", "IPv4"),
+					resource.TestCheckResourceAttr(resourceName, "nat_ip_configuration.1.private_ip_address", "10.5.1.18"),
+					resource.TestCheckResourceAttr(resourceName, "nat_ip_configuration.1.private_ip_address_version", "IPv4"),
+					resource.TestCheckResourceAttr(resourceName, "load_balancer_frontend_ip_configuration_ids.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "load_balancer_frontend_ip_configuration_ids.0"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.env", "test"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMPrivateLinkServiceExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Private Link Service not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		client := testAccProvider.Meta().(*ArmClient).Network.PrivateLinkServiceClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		if resp, err := client.Get(ctx, resourceGroup, name, ""); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Private Link Service %q (Resource Group %q) does not exist", name, resourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on network.PrivateLinkServiceClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMPrivateLinkServiceDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).Network.PrivateLinkServiceClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_private_link_service" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name, ""); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on network.PrivateLinkServiceClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMPrivateLinkService_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_private_link_service" "test" {
+  name                           = "acctestpls-%d"
```

Comment:
Could we add some capitals into the name here?
```suggestion
  name                           = "acctestPLS-%d"
```

Reply:
Fixed.